### DISCUSSION TITLE
Get local ip address after thread/network is setup

### DISCUSSION THREAD

**Discussion by nandra** [openthread/openthread#10601] [Q&A] [answered]
Hi,

I'm trying to read local IP adresses of node when is attached to thread network using `otIp6GetUnicastAddresses` but I'm always getting NULL when added to main. Is there some machanism to wait until everything is setup and get notification that I can start using network because node is setup? Thanks.

**Comment by abtink** [comment, marked as answer]
Some related APIs and mechansims to check:

- `otStateChangedCallback ` mechasnim for event notifications:
https://github.com/openthread/openthread/blob/df757ba0cf3405e5d36538f3c774500a08ff366e/include/openthread/instance.h#L190-L252

----

- `otIp6SetAddressCallback()` to be notified of address changed:

https://github.com/openthread/openthread/blob/df757ba0cf3405e5d36538f3c774500a08ff366e/include/openthread/ip6.h#L450-L483

**Comment by nandra** [comment]
Thanks for response.
